src/web3Mail/IExecWeb3Mail.ts

```diff
@@ -1,22 +1,28 @@
 import { IExec } from 'iexec';
+import { IExecConfigOptions } from 'iexec/IExecConfig';
 import { fetchMyContacts } from './fetchMyContacts.js';
-import { Contact, SendEmailParams, SendEmailResponse } from './types.js';
 import sendEmail from './sendEmail.js';
+import {
+  Contact,
+  ExternalProvider,
+  SendEmailParams,
+  SendEmailResponse,
+  Web3SignerProvider,
+} from './types.js';
 
 export class IExecWeb3Mail {
   fetchMyContacts: () => Promise<Contact[]>;
   sendEmail: (args: SendEmailParams) => Promise<SendEmailResponse>;
 
   constructor(
-    ethProvider: any,
-    { providerOptions = {}, iexecOptions = {} }: any = {}
+    ethProvider: ExternalProvider | Web3SignerProvider,
```

Comment:
`import { providers } from 'ethers';`

```suggestion
    ethProvider: providers.ExternalProvider | Web3SignerProvider,
```